### PR TITLE
Implement Select Entry

### LINKED ISSUE #145: As a student, I would like to see all bullet description of the specified entry
- command format : selectEntry INDEX
- i.e.  `selectEntry 2 ` : expands or contracts the description of the entry at index 2 shown in the displayed entry list from executing `listEntry`
- implementation subtasks
  - Model: 
      1. edit EntryDescription::toString() : returns an indexed string describing all bullets
      2. Create `TypicalEntryDescription` to return a typical `EntryDescription`
  - Logic: 
      1. logic.commands: `SelectEntryCommand` : execute() raises JumpToEntryListRequestEvent
      2.  logic.parser: create  `SelectEntryCommandParser` allow parsing of user input
      3. AddressBookParser: add case to take in selectEntry command input from user
    
  - commons:  
     1. add Event classes `UpdateExpandedEntryRequestEvent`(added to event listener of EntryListPanel and raised by addBullet) + `JumpToEntryListRequestEvent`(raised by SelectEntry)
  - UI: 
     1. ui: create classes to encapsulate the ui panel `ExpandedEntry` + `ExpanedEntry.fxml`
     2. MainWindow : a. modify layout by adding ExpandedEntryPanel inside a vbox b. update style for newly added ExpandedEntryPanel
     3. `EntryListPanel` : add event listener to selectionChangedEvent, and invokes UpdateExpandedEntryRequestEvent 

  - docs: modify UG and DG 
  - Possible enhancement:
       - AddBulletCommand: make ExpandedEntryPanel responsive to AddBulletCommand (issue #201)

## Implementation

=======
method1: Using functionality of existing personlist panel to retrieve entrycard selected
Implementation idea(with listview): based on how select command works(which will raise event in PersonListPanel) + how PersonListPanel respond to changes in model
- UI: 
1. create ExpandedEntryPanel(extending UI<Region> contains fxml listview, which contains listcell, and is connected to ObservableList<Entry> in expandedEntryManager  )
2. fxml markup file for 1

- Logic (which contains reference to model): 
1. just as `Select` , `SelectEntry` raises `JumpToEntryListRequestEvent` which will be handled by EntryListPanel, which raises another ExpanedEntryChangedEvent, instead of handled by browserPanel, is it to be handled by  ExpandedEntryPanel ---> this is the benefit of listview, it is automatically in sync with observableList<Entry > and is able to listen to selection changed event and return the reference to entry selected

- Model: 
 1. expandedEntryManager : encapsulates info needed by UI to display expanded entry
==============
Alternative method to view fully displayed entry
- command format: select INDEX
- another separate panel to display selected entry upon selection. Do away with the mouse selection, instead, using select command.
- Implementation idea:
   -  Model ExpandedEntryManager: encapsulates the current entry-to-be displayed: use filteredEntryList/ ObservableList<ResumeEntry>, which should point to the reference of ResumeEntry at a specific index--> except that the entry just has 1 element, whenever `select ` is executed, it will be updated
   -  logic(similar to how deletePerson is handled and `SelectCommand`) : parse `select INDEX`  to SelectCommand which will link to 
   - UI: create a separate class `expandedEntryList` extending UiPart<Region> which connects to ExpandedEntryManager under model to encapsulate fully displayed entry. To connect it to the model manager. -> can actually simulate the existing EntryListPanel which contains `@fxml ListView<ResumeEntry>` which already provides connection mechanism to `model.filteredEntryList`. Or can just use a `@fxml TextArea` in `ResultDisplay`
   - possible resources for reference: http://tutorials.jenkov.com/javafx/fxml.html#fxml-event-handlers

========
###Implementation idea(with listview): based on how select command works(which will raise event in PersonListPanel) + how PersonListPanel respond to changes in model
- UI: 
1. create ExpandedEntryPanel(extending UI<Region> contains fxml listview, which contains listcell, and is connected to ObservableList<Entry> in expandedEntryManager  )
2. fxml markup file for 1

- Logic (which contains reference to model): 
1. just as Select Command, raise jumpToList which will be handled by personlistpanel
2. updates expandedEntryManager ExpandedEntryPanel automatically responds to changes in ObeservableList of expandedEntryManager

- Model: 
 1. expandedEntryManager : encapsulates info needed by UI to display expanded entry

=======
Implementation idea(with textarea,): based on how select command works + how ResultDisplay responds to NewResultAvailable events(need to handle Event explicitly) raised by Command  
- UI: 
1. create ExpandedEntryPanel(extending UI<Region> contains fxml textarea )
2. fxml markup file for 1

- Logic (which contains reference to model): 
1. just as Select Command, raise jumpToList which will be handled by personlistpanel
2. raise expandedEntryChangedEvent(contain all bullet descriptions in strings) and handled by ExpandedEntryPanel

- Model: 
 1. expandedEntryManager : encapsulates info needed by UI to display expanded entry

==============

=======
- enhancement idea:
   - allowing the selection of multiple entries??

### PR DESCRIPTION
**Pull request by jhengy**
- fixes #145 
- format : selectEntry INDEX
-  After this PR, a user is able to view the fully displayed entry at a particular index shown in the displayed EntryList.
- sorry if the PR is a bit large. But the commits are mainly following the order : Model changes -> Logic Changes -> UI changes
- please refer to the associated issue for more details

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -67,6 +68,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case SelectCommand.COMMAND_WORD:
```
Comment (anubh-v) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Can we start deleting the commands used in AB4?
This will be good, especially with the bug testing event on Friday.
I suggest to do it in a separate PR

File: src/test/java/seedu/address/testutil/TypicalEntryDescription.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package seedu.address.testutil;
+
+import seedu.address.model.entry.EntryDescription;
+
+/**
+ * a class to get typical entry description.
+ */
+public class TypicalEntryDescription {
+    // Description for an entry under category ~work
+    public static final String FINANCIAL_HACK_AWARD = "Attained 2011 est Performance Award";
```
Comment (anubh-v) [inline src/test/java/seedu/address/testutil/TypicalEntryDescription.java]: Typo

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 1 above)
Comment (jhengy) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Umm, the thing is it breaks the systemtest, I still havn't be able to find a way to re-write or modify the existing system test
